src/ragger/backend/speculos.py

```diff
@@ -187,26 +192,34 @@ def compare_screen_with_snapshot(self,
             return screenshot_equal(f"{golden_snap_path}", snap)
 
     def get_current_screen_content(self) -> dict:
-        self._screen_content = self._retrieve_client_screen_content()
-        return self._screen_content
+        return self._retrieve_client_screen_content()
 
     def compare_screen_with_text(self, text: str) -> bool:
         return text in dumps(self._retrieve_client_screen_content())
 
     def wait_for_screen_change(self, timeout: float = 10.0) -> None:
         start = time()
-        content = self._retrieve_client_screen_content()
-        while content == self._screen_content:
+        screenshot = BytesIO(self._client.get_screenshot())
+        while screenshot_equal(screenshot, self._last_screenshot):
             # Give some time to other threads, and mostly Speculos one
             sleep(0.2)
             if (time() - start > timeout):
                 raise TimeoutError("Timeout waiting for screen change")
-            content = self._retrieve_client_screen_content()
+            screenshot = BytesIO(self._client.get_screenshot())
 
         # Speculos has received at least one new event to redisplay the screen
         # Wait a bit to ensure the event batch is received and processed by Speculos before returning
         sleep(0.2)
```

Comment:
I don't really known. I'd say it depend on how the screen is refreshed. It could be done in multiple part and we would have still the same possible race issue?

Reply:
Ok then let's keep it I'm also not sure it's safe to remove